### ISSUE TITLE
fix(file-resolver): generalImport() prepends current Windows drive to POSIX absolute paths

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3840]
- Markuplint version: dev branch HEAD (v5 RC)
- Parser lang: N/A (file-resolver / generalImport)
- Node.js version: 24
- OS: Windows

## Describe the bug

`packages/@markuplint/file-resolver/src/general-import.ts:40-44` calls `pathToFileURL(name).href` for any path that satisfies `path.isAbsolute(name)`.

On a Windows runtime, `path.isAbsolute('/tmp/foo')` returns `true` (POSIX absolute paths are accepted as a "rooted" path on Windows), and `pathToFileURL('/tmp/foo')` resolves the path against the **current drive**, producing `file:///D:/tmp/foo` instead of `file:///tmp/foo`.

This is the same root cause as #3836 (vscode/server side, fixed for v4 in the sibling PR). The dev branch carries the identical pattern in `general-import.ts`, so any caller that passes a POSIX-style absolute path to `generalImport()` on Windows will load the wrong file (or fail to load it). Realistic call sites: loading configs / parsers / plugins by absolute path, especially in WSL or remote-development setups where POSIX paths can reach a Windows Node process.

## Code Example or Playground URL

```ts
// packages/@markuplint/file-resolver/src/general-import.ts (dev)
if (path.isAbsolute(name)) {
  // Use Node.js pathToFileURL function to convert to a proper URL
  importPath = pathToFileURL(name).href;  // ← misbehaves on Windows for POSIX paths
  gLog('Converted to file URL: %s', importPath);
}
```

## Steps To Reproduce

On Windows:

```ts
import { pathToFileURL } from 'node:url';
console.log(pathToFileURL('/tmp/foo.mjs').href);
// → 'file:///D:/tmp/foo.mjs'  (current drive prepended)
// expected: 'file:///tmp/foo.mjs'
```

`generalImport('/tmp/foo.mjs')` then calls `import('file:///D:/tmp/foo.mjs')`, loading the wrong file or surfacing `ERR_MODULE_NOT_FOUND`.

## The current behavior

POSIX absolute paths on Windows are silently rewritten to point under the current drive (`D:/tmp/foo.mjs`).

## The expected behavior

POSIX absolute paths should round-trip: `/tmp/foo.mjs` → `file:///tmp/foo.mjs`, regardless of the host OS. This matches the contract that the v4 PR for #3836 establishes for the parallel `vscode/src/server/get-module.ts` implementation.

## Relationship to #3836

- #3836 covers the same bug in `vscode/src/server/get-module.ts`. The vscode wrapper (`toImportSpecifier()`) already short-circuits POSIX absolute paths via explicit URL construction; file-resolver does not.
- The vscode source even contains a JSDoc instruction to "Mirrors `packages/@markuplint/file-resolver/src/general-import.ts` — keep the two in sync when adjusting Windows-path handling." That mirror was never enforced, so file-resolver retained the pre-#3807 buggy approach.

## Suggested fix direction

Replace the `pathToFileURL` call with the same OS-independent explicit URL construction used by `toImportSpecifier()` in `vscode/src/server/get-module.ts`:

```typescript
if (path.isAbsolute(name)) {
  const isWindowsAbsolute = /^[a-z]:[/\\]/i.test(name);
  if (isWindowsAbsolute) {
    const [drive, ...rest] = name.replaceAll('\\', '/').split('/');
    const encoded = [drive, ...rest.map(encodeURIComponent)].join('/');
    importPath = `file:///${encoded}`;
  } else {
    const segments = name.split('/').map(encodeURIComponent);
    importPath = `file://${segments.join('/')}`;
  }
}
```

Or extract a shared helper (e.g. `toFileURL()` in `@markuplint/shared` or `packages/@markuplint/file-resolver/src/path-utils.ts`) that vscode and file-resolver both consume.

## Tests

`generalImport()` has no spec file currently. The fix should add a spec that covers POSIX, Windows, encoding, and the OS-independent guarantee.
